src/test/java/com/example/springboot/controller/BookControllerTest.java

```diff
@@ -0,0 +1,252 @@
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class BookControllerTest {
+    protected static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeEach
+    void beforeEach(
+            @Autowired DataSource dataSource,
+            @Autowired WebApplicationContext applicationContext
+    ) throws SQLException {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+        teardown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/add-two-books-to-books-table.sql")
+            );
+        }
+    }
+
+    @AfterEach
+    void afterEach(@Autowired DataSource dataSource) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/remove-two-books-from-books-table.sql")
+            );
+        }
+    }
+
+    @WithMockUser(username = "user", roles = {"USER"})
+    @Test
+    @DisplayName("Get all books")
+    void getAll_GivenBooksInCatalog_ReturnsAllBooks() throws Exception {
+        BookDto bookDto2 = createBookDto();
+        bookDto2.setTitle("The Lord of The Rings");
+        bookDto2.setAuthor("Tolkien");
+        bookDto2.setIsbn("12345687");
+        bookDto2.setPrice(BigDecimal.valueOf(150));
+        bookDto2.setDescription("Story about rings of the power");
+        BookDto bookDto1 = createBookDto();
+
+        List<BookDto> expected = new ArrayList<>();
+        expected.add(bookDto1);
+        expected.add(bookDto2);
+        String params = "?page=0&size=2";
+
+        MvcResult result = mockMvc.perform(
+                get("/books" + params)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto[] actual = objectMapper.readValue(result.getResponse()
+                .getContentAsByteArray(), BookDto[].class);
+        Assertions.assertEquals(expected.size(), actual.length);
+        Assertions.assertEquals(expected.get(0).getTitle(), actual[0].getTitle());
```

Comment:
use andExpect + jsonPath instead of object mapper. 
It will simplify your work

Reply:
I need use it with all tests?

Reply:
exactly, so please fix other tests